AstronomyObjC/AstronomyObjC/Controllers/TMCFetchOperations.m

```diff
@@ -0,0 +1,79 @@
+//
+//  TMCFetchOperations.m
+//  AstronomyObjC
+//
+//  Created by Christopher Devito on 5/19/20.
+//  Copyright © 2020 Christopher Devito. All rights reserved.
+//
+
+#import "TMCFetchOperations.h"
+#import "TMCMarsPhotoReference.h"
+#import "NSURL+TMCUsingHTTPS.h"
+
+@interface TMCFetchOperations();
+
+@property (nonatomic, copy, nullable) NSData* imageData;
+@property TMCMarsPhotoReference *photo;
+@property (nonatomic, readwrite) NSURLSessionDataTask *task;
+typedef NS_ENUM(int, State) {isReady, isExecuting, isFinished};
+@property State state;
+
+@end
+
+@implementation TMCFetchOperations
+- (instancetype)initWithPhotoReference:(TMCMarsPhotoReference *)photo{
+    self = [super init];
+    if (self) {
+        _photo = photo;
+        self.state = isReady;
+    }
+    return self;
+}
+
+- (BOOL)isAsynchronous {
+    return YES;
+}
+
+- (void)start {
+    self.state = isExecuting;
+    NSURL *url = self.photo.imageURL.usingHTTPS;
+    NSURLSessionTask *task = [NSURLSession.sharedSession dataTaskWithURL:url completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+
+        if (error) {
+            self.fetchError = error;
+            self.state = isFinished;
+            return;
+        }
+
+        if (!data) {
+            self.state = isFinished;
+            return;
+        }
+
+        self.imageData = data;
+        self.state = isFinished;
+        return;
+    }];
+    [task resume];
+}
+- (void)cancel {
+    [self.task cancel];
+}
+- (BOOL)isReady {
+    return self.state == isReady;
+}
```

Comment:
I would not override `isReady` read the docs ... it manages dependencies and you have a risk of not doing it correctly.

I'm not sure if all these states are considered independent, so look at the docs for these methods before you override in the `NSOperation` class